Bucketeer/Sources/Internal/Remote/ApiClientImpl.swift

```diff
@@ -11,6 +11,7 @@ final class ApiClientImpl: ApiClient {
     private let defaultRequestTimeoutMills: Int64
     private let logger: Logger?
     private let semaphore = DispatchSemaphore(value: 0)
+    private var closed = false
```

Comment:
Is this thread-safe?

Reply:
yes, because we are accessing the APIClient in the SDK queue. It's safe if we keep doing this.
So it could be a simple bool variable.
Let me know if you think we need to make it more sure  @cre8ivejp

Reply:
As long we call the APIs using the same SDK queue, it's fine.
Thanks for confirming.